common/persistence/sql/sqlplugin/postgresql/nexus_incoming_services.go

```diff
@@ -25,47 +25,129 @@ package postgresql
+	getIncomingServicesQry   = `SELECT service_id, data, data_encoding, version FROM nexus_incoming_services WHERE service_id > $1 ORDER BY service_id LIMIT $2`
+)
+
 func (pdb *db) InitializeNexusIncomingServicesTableVersion(ctx context.Context) error {
-	return serviceerror.NewUnimplemented("InitializeNexusIncomingServicesTableVersion is not implemented for PostgreSQL plugin")
+	result, err := pdb.conn.ExecContext(ctx, createIncomingServicesTableVersionQry)
+	if err != nil {
+		return err
+	}
+	nRows, err := result.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if nRows != 1 {
+		return persistence.ErrNexusTableVersionConflict
+	}
+	return nil
 }
 
 func (pdb *db) IncrementNexusIncomingServicesTableVersion(
 	ctx context.Context,
 	lastKnownTableVersion int64,
 ) error {
-	return serviceerror.NewUnimplemented("IncrementNexusIncomingServicesTableVersion is not implemented for PostgreSQL plugin")
+	result, err := pdb.conn.ExecContext(ctx, incrementIncomingServicesTableVersionQry, lastKnownTableVersion+1, lastKnownTableVersion)
+	if err != nil {
+		return err
+	}
+	nRows, err := result.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if nRows != 1 {
+		return persistence.ErrNexusTableVersionConflict
+	}
+	return nil
+}
+
+func (pdb *db) GetNexusIncomingServicesTableVersion(
+	ctx context.Context,
+) (int64, error) {
+	version := int64(-1)
+	err := pdb.conn.GetContext(ctx, &version, getIncomingServicesTableVersionQry)
+	if errors.Is(err, sql.ErrNoRows) {
+		return 0, nil
+	}
+	return version, err
 }
 
 func (pdb *db) InsertIntoNexusIncomingServices(
 	ctx context.Context,
 	row *sqlplugin.NexusIncomingServicesRow,
 ) error {
-	return serviceerror.NewUnimplemented("InsertIntoNexusIncomingServices is not implemented for PostgreSQL plugin")
+	result, err := pdb.conn.ExecContext(
+		ctx,
+		createIncomingServiceQry,
+		row.ServiceID,
+		row.Data,
+		row.DataEncoding)
+	if err != nil {
+		return err
+	}
+	nRows, err := result.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if nRows != 1 {
+		return persistence.ErrNexusTableVersionConflict
+	}
+	return err
 }
 
 func (pdb *db) UpdateNexusIncomingService(
 	ctx context.Context,
 	row *sqlplugin.NexusIncomingServicesRow,
 ) error {
-	return serviceerror.NewUnimplemented("UpdateNexusIncomingService is not implemented for PostgreSQL plugin")
-}
+	result, err := pdb.conn.ExecContext(
+		ctx,
+		updateIncomingServiceQry,
+		row.Data,
+		row.DataEncoding,
+		row.Version+1,
+		row.ServiceID,
+		row.Version)
+	if err != nil {
+		return err
+	}
 
-func (pdb *db) ListNexusIncomingServices(
-	ctx context.Context,
-	request *sqlplugin.ListNexusIncomingServicesRequest,
-) (*sqlplugin.ListNexusIncomingServicesResponse, error) {
-	return nil, serviceerror.NewUnimplemented("ListNexusIncomingServices is not implemented for PostgreSQL plugin")
+	nRows, err := result.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if nRows != 1 {
+		return persistence.ErrNexusIncomingServiceVersionConflict
```

Comment:
I _think_ you should be able to move this (and similar locations) `RowsAffected` logic to the sql plugin instead of duplicating in all of the different implementations.